src/ZEO/ClientStorage.py

```diff
@@ -799,40 +796,35 @@ def tpc_begin(self, txn, tid=None, status=' '):
             raise POSException.ReadOnlyError()
 
         try:
-            tbuf = txn.data(self)
+            tbuf = txn.__buf
         except AttributeError:
-            # Gaaaa. This is a recovery transaction. Work around this
```

Comment:
Recovery transactions are TransactionRecords, right? Perhaps it would be cleaner to give them the same data/set_data interface as a Transaction has? In the same way that we unified the extension/extended_info attribute between the two.